.github/workflows/ci.yaml

```diff
@@ -0,0 +1,73 @@
+name: Continuous integration
+on: [ push, pull_request ]
+
+# TODO: fix tests on x86_64-linux, run cargo test + cargo taurpaulin + codecov upload.
+
+env:
+  BINDGEN_EXTRA_CLANG_ARGS: -B/usr/include/
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - beta
+          - nightly
+          - 1.60.0  # MSRV
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - uses: actions/cache@v3
+        with:
+          path: |
+                target/xtask/
+          key: ${{ runner.os }}-xtask
+
+      - run: sudo apt-get install -y avr-libc tar wget
+
+      - name: Install `arm-frc2022-unknown-gcc` toolchain.
+        run: wget https://github.com/wpilibsuite/roborio-toolchain/releases/download/v2022-1/FRC-2022-Linux-Toolchain-7.3.0.tar.gz && tar -xzvf FRC-2022-Linux-*.tar.gz && echo "$(pwd)/frc2022/roborio/bin/" >> $GITHUB_PATH
```

Comment:
I'm surprised this isn't using the Nix closure?

Reply:
Good point, but installing Mix takes *a lot* of time in GH actions. It would be easier to eventually write a test for the Nix development environment, too.

Reply:
It doesn't look like installing Nix in GitHub Actions takes long at all? https://github.com/cachix/install-nix-action

Reply:
I meant during runtime :) Nix is a huge package manager that takes up a few gigabytes during installation. And the nix store is even larger - on my machine, it's already 50 Gigs.  It would be a waste of CI time to do this with Nix. But, since you're the administrator on this project, if you feel it's necessary, no problem.